### PR TITLE
Issue 61 : Backend Add details in the datasource card

### LINKED ISSUE #61: Add datasource details in the list
When listing data sources, it will be useful to see some details like the cluster identifier, the database or the user as a description:

unnamed

Similar to postgresql:

unnamed (2)

### PR DESCRIPTION
**Pull request by yaelleC**
Backend work for #61 

## Description
When listing datasources, a card is composed of a Heading ('Amazon Redshift'), a Figure (the logo) and some `meta` info. Meta include the datasource `typeName`, `url` and `isDefault`.

This PR creates a new endpoint for getting a cluster url 

A next PR #127 will use it to update the description on the datasource card

## Test coverage

_**Current Redshift coverage for backend is: 48.6%**_

```
/pkg/redshift/api/api.go:345:           Cluster                         100.0%
/pkg/redshift/datasource.go:144:          (all)                         0.0%
/pkg/redshift/routes/routes.go:36:        cluster                         57.1%
/pkg/redshift/routes/routes.go:47:         Routes                         100.0%

total:                                    (statements)                51.4%
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,8 @@
   "scripts": {
     "build": "grafana-toolkit plugin:build",
     "test": "jest --notify --watch",
+    "test:coverage": "jest --coverage",
+    "test:coverage:changes": "jest --coverage --changedSince=origin/main",
```
Comment (yaelleC) [inline package.json:9]: I find these commands quite handy... But they are not needed for this PR, 
happy to make this it's own PR if preferred?

File: src/ConfigEditor/ConfigEditor.tsx [hunk 2]
```diff
@@ -72,6 +72,18 @@ export function ConfigEditor(props: Props) {
...
   }, [arn]);
 
+  // Description to show in datasource card in list
+  const { dbUser, clusterIdentifier, database } = props.options.jsonData || {};
+  useEffect(() => {
+    console.log(`${dbUser}@${clusterIdentifier}/${database}`);
+    if (dbUser || clusterIdentifier || database) {
+      props.onOptionsChange({
+        ...props.options,
+        url: `${dbUser}@${clusterIdentifier}/${database}`,
```
Comment (yaelleC) [inline src/ConfigEditor/ConfigEditor.tsx:82]: As mentioned in PR desc, two questions here:

1. Is it okay to reuse the `url` field? A better way could be to create a details optional prop in the Grafana `DataSourceSettings` interface and use that instead to make it clear we are not restricted to URLs, but that seemed like a big change... thoughts?

2. Also happy to change the formatting (`user@cluster/db`) if you think of something better, one concern could be if one of the three is not populated we will have something like `user@/db`. Maybe we could go for a simpler `user: testUser, cluster: testCluster, database: testDB` ?

Comment (andresmgot) [reply src/ConfigEditor/ConfigEditor.tsx:82]: I think we can make this a real URL, similar to what we do for PostgreSQL. If you go to the cluster settings in the aws console ([link](https://us-east-2.console.aws.amazon.com/redshiftv2/home?region=us-east-2#cluster-details?cluster=redshift-cluster-grafana)), you can see an Endpoint there that looks like: `cluster.stuff....com:5439/db`. We can try and show the same information here. It requires some backend changes though.

The target would be to use this API: https://docs.aws.amazon.com/redshift/latest/APIReference/API_DescribeClusters.html. This is the one that has the `Endpoint` info that we want. To achieve this, we need:

- Add a new Client in the API struct: https://github.com/grafana/redshift-datasource/blob/main/pkg/redshift/api/api.go#L54. At the moment, we only have a `redshiftdataapiservice`  and `secretsmanager` clients, but the target API method (`DescribeClusters`) is in the `redshift` API so we need to add a third client. 
- Add a new method to get the cluster endpoint. It can look very similar to the [Secret](https://github.com/grafana/redshift-datasource/blob/main/pkg/redshift/api/api.go#L321) method but internally using the DescribeClusters method instead.
- Extend the `RedshiftDatasourceIface` with the method we just added and create the function that just calls to the api function (again, just like `Secret`).
- Add a resource route (as with the [secret](https://github.com/grafana/redshift-datasource/blob/main/pkg/redshift/routes/routes.go#L25)). This will be the endpoint our frontend will hit to get the information.

Those are all the backend steps! now for the frontend, we just need to fetch that URL as we do for the secret and save that info here.

If you want to pair for this one, feel free to schedule some time :)

File: pkg/redshift/api/api.go [hunk 3]
```diff
@@ -337,3 +341,31 @@ func (c *API) Secret(ctx aws.Context, options sqlds.Options) (*models.RedshiftSe
...
+	}
+	if out == nil {
+		return nil, fmt.Errorf("missing cluster content")
+	}
+	res := &models.RedshiftCluster{}
+	for _,r := range out.Clusters {
+		if (r != nil && r.ClusterIdentifier != nil && *r.ClusterIdentifier == clusterId) {
+			res.Endpoint = models.RedshiftEndpoint{
+				Address: *r.Endpoint.Address,
+				Port: *r.Endpoint.Port,
```
Comment (andresmgot) [inline pkg/redshift/api/api.go:362]: we need also checks that verify that Address and Port are not `nil`

File: pkg/redshift/api/api_test.go [hunk 4]
```diff
@@ -216,3 +216,51 @@ func Test_GetSecret(t *testing.T) {
...
+	}
+	if !cmp.Equal(expectedCluster, cluster) {
+		t.Errorf("unexpected result: %v", cmp.Diff(expectedCluster, cluster))
+	}
+}
+
+func Test_GetCluster_Errors(t *testing.T) {
+	fooC := &API{ManagementClient: &redshiftclientmock.MockRedshiftClient{Clusters: []string{"foo"}}}
+
+	tests := []struct {
```
Comment (andresmgot) [inline pkg/redshift/api/api_test.go:241]: extra points if you merge this test with the previous one :) (making `expectedCluster` and the error optional).

File: pkg/redshift/api/api_test.go [hunk 5]
```diff
@@ -216,3 +216,51 @@ func Test_GetSecret(t *testing.T) {
...
+			desc: "Error cluster ID not found",
+			clusterId: "xyz",
+			errMsg: "ClusterId xyz not found",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			c := tt.c
+			cluster, err := c.Cluster(sqlds.Options{"clusterIdentifier": tt.clusterId})
+			if (err.Error() != tt.errMsg) {
```
Comment (andresmgot) [inline pkg/redshift/api/api_test.go:258]: `err` may be `nil` too, so you need to check that to avoid a panic in the test.

Comment (andresmgot) [reply pkg/redshift/api/api_test.go:258]: also, you can simplify these checks with the [`assert` package](https://pkg.go.dev/github.com/stretchr/testify/assert#EqualError):

```
			assert.Nil(t, cluster)
			assert.EqualError(t, err, tt.errMsg)
```

will do what you want

**Review by andresmgot — COMMENTED** [review]
Good job here! I just have some minor comments.

File: pkg/redshift/api/api_test.go (hunk 4 above)
Comment (yaelleC) [reply pkg/redshift/api/api_test.go:241]: Yes that was the plan! 
Still trying to figure out a test for the other errors too 😉

**Comment by yaelleC** [comment]
> btw, don't forget to update the docs with the minimal policy adding the new required permission

I've updated the readme, anything else I should change?

**Review by andresmgot — APPROVED** [review]
LGTM!

**Comment by andresmgot** [comment]
> > btw, don't forget to update the docs with the minimal policy adding the new required permission
> 
> @andresmgot : I've updated the readme, anything else I should change?

Nope, that's enough. Thanks!
